File: opcua-webservice-spring/core/src/main/java/org/opcfoundation/webserver/types/message/digitaltwin/GetDescriptorResponse.java
package org.opcfoundation.webserver.types.message.digitaltwin;

import org.eclipse.milo.opcua.stack.core.types.builtin.LocalizedText;

public class GetDescriptorResponse {
    private final LocalizedText displayName;
    private final LocalizedText description;

    public GetDescriptorResponse(String displayName)
    {
        this.displayName = new LocalizedText(displayName);
        this.description = LocalizedText.NULL_VALUE;
    }

    public GetDescriptorResponse(LocalizedText displayName, LocalizedText description)
    {
        this.displayName = displayName;
        this.description = description;
    }

    public LocalizedText getDisplayName() {
        return displayName;
    }

    public LocalizedText getDescription() {
        return description;
    }
}

File: opcua-webservice-node/source/src/server/service/message/digitialtwin/GetDescriptorResponse.ts
import { UaLocalizedText } from "opcua-webapi-ts";

export class GetDescriptorResponse {
    private readonly _displayName: UaLocalizedText;
    private readonly _description: UaLocalizedText;

    constructor(displayName: string);
    constructor(displayName: UaLocalizedText, description: UaLocalizedText);
    constructor(displayName: string | UaLocalizedText, description: UaLocalizedText = UaLocalizedText.nullText) {
        this._displayName = typeof displayName === "string"
            ? new UaLocalizedText(displayName)
            : displayName;
        this._description = description;
    }

    get displayName(): UaLocalizedText {
        return this._displayName;
    }

    get description(): UaLocalizedText {
        return this._description;
    }
}